qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -90,6 +112,108 @@ def quantum_instance(
         else:
             self._quantum_instance = quantum_instance
 
+    @property
+    def user_parameters(self) -> Union[ParameterVector, Sequence[Parameter]]:
+        """Return the vector of user parameters."""
+        return self._user_parameters
+
+    @user_parameters.setter
+    def user_parameters(self, user_params: Union[ParameterVector, Sequence[Parameter]]) -> None:
+        """Sets the user parameters"""
+        if user_params:
```

Comment:
This `if` statement implies that `user_params` may be `None`. 
So, two things:

- If `None` is possible, then perhaps the type hint should be `Optional[Union[ParameterVector, Sequence[Parameter]]]`?
- Better to have an explicit test for `None`: `if user_params is not None:`

Am I missing anything here? I suspect that `user_params` should not be `None`.

Reply:
The logic is a little funny here. If users want to use `QuantumKernel` in the traditional way (no user parameters), this field will stay `None`; however the setter is always called in the constructor, so I need to handle the 'None' case.

I will take your advice of explicitly handling the `None` case.

Reply:
> I will take your advice of explicitly handling the `None` case.

OK, thanks. Looks good.